VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java

```diff
@@ -54,27 +66,48 @@ public void call(Session session, SessionState state, Exception exception) {
         }
     };
 
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    ProfileModel profile;
+
+    ImageView imageView;
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         uiLifecycleHelper = IUiLifecycleHelper.Factory.getInstance(getActivity(), callback);
         uiLifecycleHelper.onCreate(savedInstanceState);
-    }
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-            Bundle savedInstanceState) {
-        super.onCreateView(inflater, container, savedInstanceState);
         Context context = getActivity().getBaseContext();
-
         socialPref = new PrefManager(context, PrefManager.Pref.FEATURES);
-
         pref = new PrefManager(context, PrefManager.Pref.LOGIN);
+        profile = pref.getCurrentUserProfile();
+        if (config.isUserProfilesEnabled() && profile != null && profile.username != null) {
+            getAccountTask = new GetAccountTask(getActivity(), profile.username) {
+                @Override
+                protected void onSuccess(Account account) throws Exception {
+                    final ProfileImage profileImage = account.getProfileImage();
+                    Glide.with(NavigationFragment.this)
+                            .load(profileImage.getImageUrlLarge())
+                                    .skipMemoryCache(true)
+                            .into(imageView);
```

Comment:
We should check that `imageView` is not `null`, and also store the data locally so that it can be used from `onCreateView()` as well. Also we might want to store this into the existing profile data (thereby refreshing it as well).